### PR TITLE
[SDK] Replace NSRange.toRange() with failable initialiser on Range<Int>

### PR DESCRIPTION
**Pull request by karwa**
Fixes ABI FIXME #75

### CODE REVIEW

File: stdlib/public/SDK/Foundation/NSRange.swift [hunk 1]
```diff
@@ -16,18 +16,22 @@
...
 extension NSRange {
   public init(_ x: Range<Int>) {
     location = x.lowerBound
     length = x.count
   }
 
-  // FIXME(ABI)#75 (Conditional Conformance): this API should be an extension on Range.
-  // Can't express it now because the compiler does not support conditional
-  // extensions with type equality constraints.
+  @available(*, deprecated, message: "Use Range(_: NSRange) initialiser instead")
```
Comment (slavapestov) [inline stdlib/public/SDK/Foundation/NSRange.swift]: "initializer"

**Comment by parkera** [comment]
cc @phausler

File: stdlib/public/SDK/Foundation/NSRange.swift (hunk 1 above)
Comment (parkera) [inline stdlib/public/SDK/Foundation/NSRange.swift]: Please make sure to get review from someone from Foundation for changes in the overlay.

**Comment by parkera** [comment]
Please make sure to get review from someone from Foundation for changes in the overlay.

**Comment by karwa** [comment]
Updated a comment on String.UTF16View

Sure, I'll make sure to request a review from one of you next time :). I just happened to notice that this ABI FIXME was implementable now.

**Comment by karwa** [comment]
pinging @phausler

**Comment by phausler** [comment]
This is an API change and would make swift-corelibs deviate from the Darwin versions

**Comment by parkera** [comment]
This is in the overlay, actually. I think we should consider it for post-Swift 4.

**Comment by karwa** [comment]
The ABI FIXME led me to believe this source-breaking change was always planned when the requisite compiler features were available. It's just about when we force users to change their code. If the design is settled, I'd advocate for getting the "definitive" API out there as soon as possible.

**Comment by tkremenek** [comment]
This pull request was opened 22 months ago.  Is this still worth keeping open?  There has been no movement here, and this PR cannot be merged as is into `master`.

**Comment by CodaFi** [comment]
The bulk of this has been subsumed by the new `Range(_:in:)` constructor.  I'm going to close this.  If there is any more movement or you would like to revisit this, please reopen this pull request and rebase it onto master @karwa.

**Pull request closed without merging**